File: Learning-material/04-testing/ejemplos/carrito.js
function subtotal(items) {
  return items.reduce((acc, [precio, cant]) => acc + precio * cant, 0);
}
function aplicarDescuento(monto, porcentaje) {
  if (porcentaje < 0 || porcentaje > 100) throw new Error("porcentaje invalido");
  return monto * (1 - porcentaje / 100);
}
module.exports = { subtotal, aplicarDescuento };

File: Learning-material/04-testing/ejemplos/carrito.py
def subtotal(items):
    """items: lista de (precio, cantidad). Devuelve el subtotal."""
    return sum(precio * cant for precio, cant in items)

def aplicar_descuento(monto, porcentaje):
    if not 0 <= porcentaje <= 100:
        raise ValueError("porcentaje debe estar entre 0 y 100")
    return monto * (1 - porcentaje / 100)
